code.rb

```diff
@@ -0,0 +1,150 @@
+
+  def my_select
+    if block_given?
+      arr = self
+      result = []
+      0.upto(arr.length - 1) do |i|
+        result << arr[i] if yield(arr[i])
+      end
+      result
+    else
+      arr.to_enum
+    end
+  end
+
+  def my_all?(arg = nil)
+    resp = true
+    return true if to_a.nil? || (self.class.to_s == 'Hash' && !block_given?)
+
+    if block_given?
+      to_a.my_each { |val| resp = false unless yield(val) }
+    elsif arg.nil?
+      to_a.my_each { |val| resp = false unless val }
+    else
+      case arg.class.to_s
+      when 'Regexp'
+        to_a.my_each { |val| resp = false unless arg.match? val.to_s }
+      when 'Class'
+        to_a.my_each { |val| resp = false unless val.is_a? arg }
+      else
+        to_a.my_each { |val| return resp = false unless val == arg }
+      end
+    end
+    resp
+  end
+
+  def my_any?(arg = nil)
+    resp = false
+    return false if to_a.nil?
+    return true if self.class.to_s == 'Hash' && !block_given?
+
+    if block_given?
+      to_a.my_each { |val| resp = true if yield(val) }
+    elsif arg.nil?
+      to_a.my_each { |val| resp = true if val }
+    else
+      case arg.class.to_s
+      when 'Regexp'
+        puts 'regex'
+        to_a.my_each { |val| resp = true if arg.match? val.to_s }
+      when 'Class'
+        puts 'class'
+        to_a.my_each { |val| resp = true if val.is_a? arg }
+      else
+        puts 'else'
+        to_a.my_each { |val| return resp = true if val == arg }
+      end
+    end
+    resp
+  end
+
+  def my_none?(arg = nil)
+    resp = true
+    return true if to_a.nil?
+    return false if self.class.to_s == 'Hash' && !block_given?
+
+    if block_given?
+      to_a.my_each { |val| resp = false if yield(val) }
+    elsif arg.nil?
+      to_a.my_each { |val| resp = false if val }
+    else
+      case arg.class.to_s
+      when 'Regexp'
+        to_a.my_each { |val| resp = false if arg.match? val.to_s }
+      when 'Class'
+        to_a.my_each { |val| resp = false if val.is_a? arg }
+      else
+        to_a.my_each { |val| return resp = false if val == arg }
+      end
+    end
+    resp
+  end
+
+  def my_count(arg = nil)
+    arr = self
+    return arr.length unless block_given? || arg
+
+    counter = 0
+    if arg
+      arr.my_each { |val| counter += 1 if val == arg }
+    else
+      arr.my_each { |val| counter += 1 if yield(val) }
+    end
+    counter
+  end
+
+  def my_map(a_proc = nil)
+    arr = self
+    return to_enum(:my_map) unless block_given?
+
+    if a_proc.nil?
+      arr.my_each_with_index { |v, i| arr[i] = yield(v) }
+    else
+      arr.my_each_with_index { |v, i| arr[i] = a_proc.call(v) }
+    end
+    arr
+  end
+
+  def my_inject(init = to_a[0], oper = :+)
+    if init.is_a? Symbol
+      oper = init
+      init = to_a[0]
+    end
+    accum = init
+    if block_given?
+      to_a.my_each { |val| accum = yield(accum, val) }
+    else
+      to_a.my_each { |val| accum = accum.send(oper, val) }
+    end
+    accum
+  end
+end
```

Comment:
* [ ] `my_inject` method should be identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- [ ]  Enumerables `my_inject` combines all elements of enum by applying a binary operation, specified by a block

```ruby
[1,2,3,4,5,6].my_inject(&:+) != [1,2,3,4,5,6].inject(&:+)
```

- [ ]  Enumerables `my_inject` when a symbol is specified combines each element of the collection by applying the symbol as a named method

```ruby
[1,2,3,4,5,6].my_inject(:+) == [1,2,3,4,5,6].inject(:+)
```